zip-frost.rst

```diff
@@ -0,0 +1,403 @@
+signatures in the Zcash protocol, for the Sapling and Orchard network upgrades.
+
+
+Motivation
+==========
+
+In the Zcash protocol, Spend Authorization Signatures are employed to effectively
+authorize a transaction. The ability to generate these signatures with the user's
+private key is what effectively allows the user to spend funds.
+
+This is a security-critical step, since anyone who obtains access to the private
+key will be able to spend the user's funds. For this reason, one interesting
+possibility is to require multiple entities to allow the transaction to go through.
+This can be accomplished with threshold signatures, where... (TODO: explain)
+
+FROST is one of such threshold signatures protocols. However, it can't be used as-is
+since the Zcash protocol also requires re-randomizing public and private keys
+to ensure unlinkability between transactions. This ZIP specifies a variant of
+FROST with re-randomization support.
+
+
+Requirements
+============
+
+- All signatures generated by following this ZIP MUST be verified successfully
+  as Sapling or Orchard spend authorization signatures.
+- The signatures generated by following this ZIP should meet the security criteria
+  for Signature with Re-Randomizable Keys as specified in the Zcash protocol [#protocol]_
+- Follow the threat model described below.
+
+Threat Model
+------------
+
+In normal usage, a Zcash user follows multiple steps in order to generate a
+shielded transaction:
+
+- The transaction is created.
+- The transaction is signed with a randomized version of the user's spend
+  authorization private key.
+- The zero-knowledge proof for the transaction is created with the randomizer
+  as an auxiliary (secret) input, among others.
+
+When employing re-randomized FROST as specified in this ZIP, the goal is to
+split the spend authorization private key among multiple participants. This means
+that the proof generation will still be generated be a single participant, likely
+the one that created the transaction in the first place and that will actually
+submit the transaction to the network. Note that his user already controls the
+privacy of the transaction since they are responsible for creating the proof.
+
+This fits well into the "Coordinator" role from the FROST specification. The
+Coordinator is responsible for sending the message to be signed to all participants,
+and to aggregate the signature shares.
+
+With those considerations in mind, the threat model considered in this ZIP is:
+
+- The Coordinator is trusted with the privacy of the transaction (which includes
+  the unlinkability property). A rogue Coordinator will be able to break
+  unlinkability and privacy, but should not be able to create signed transactions
+  without the approval of `MIN_SIGNERS` participants, as specified in FROST.
+
+
+Non-requirements
+================
+
+- The control of the transaction privacy is not distributed among participants
+  and is trusted with the Coordinator.
```

Comment:
I don't believe this is effectively worded given the participants learn of the randomizer. Accordingly, every participant is also trusted not to break unlinkability. Re-defining sign to take in the randomized key is possible, removing this possibility.

Reply:
Good point, fixed in 3a96a2f